### PR TITLE
fix: harden no-throw failure paths and hook teardown safety

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary

Remediates findings from the independent audit, making failure paths in an injected-DLL context as robust as the normal paths.

- Async logger uses aligned nothrow allocation and a placement-new singleton, so logging can no longer throw out of a `noexcept` boundary.
- Input reshape APIs and bootstrap helpers are genuinely no-throw (fail-closed on out-of-memory) instead of risking `std::terminate`.
- New no-throw logging entry points (`log_noexcept`, `try_log`) for hook and input callbacks.
- HookManager now has uniform post-gate shutdown re-checks and no longer logs while holding locks.
- Raw scanner overloads document their readable-range precondition, and sanitizers fail fast on non-MSVC Windows toolchains.
- Style, naming, and formatting cleanups, plus new convention rules, README notes, and tests covering the no-throw and reshape paths.

## Summary by CodeRabbit

* **New Features**
  * Added no-throw logging APIs for critical error paths.

* **Bug Fixes**
  * Hardened memory allocation and exception handling in input/hook management.
  * Improved deferred logging patterns to prevent deadlocks.
  * Added Windows sanitizer validation at configure time.

* **Documentation**
  * Clarified C++ concurrency guidelines and Windows sanitizer requirements.
  * Updated memory scanning preconditions for safe usage.

### CODE REVIEW

File: src/memory.cpp [hunk 1]
```diff
@@ -890,22 +890,22 @@ bool DetourModKit::Memory::init_cache(size_t cache_size, unsigned int expiry_ms,
...
+                        if (s_cleanup_thread.joinable())
                         {
                             pin_current_module();
-                            s_cleanupThread.detach();
+                            s_cleanup_thread.detach();
                             DetourModKit::Diagnostics::record_intentional_leak(DetourModKit::Diagnostics::LeakSubsystem::MemoryCache);
                         }
-                        s_cacheInitialized.store(false, std::memory_order_release);
+                        s_cache_initialized.store(false, std::memory_order_release);
                         return;
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply src/memory.cpp:909]: @coderabbitai

**hook_manager.cpp -- remove_hook erase-phase miss returns success**
Fixed. The exclusive erase phase now returns `HookError::HookNotFound` (with a deferred warning) when the second `m_hooks.find()` misses due to a concurrent removal, instead of a false success. The found/erase/log success path is unchanged.

**input.cpp -- debug log under the state mutex (update_combos and update_binding_combos)**
Fixed. Both "name not found" paths now `lock.unlock()` before calling `try_log(...)`, so logging never runs while the bindings/state mutex is held. (`InputManager::update_binding_combos` was switched from `lock_guard` to `unique_lock` to allow the early release.)

**input.cpp -- recompute_modifier_caches_locked() catch leaves stale caches**
Fixed. The catch now clears `m_name_index`, `m_known_modifiers`, and the gamepad/wheel flags, and publishes an empty detour rule list, so a failed cache rebuild cannot leave name-index entries addressing past a shrunk `m_bindings`. Because `recompute_modifier_caches_locked()` runs after every reshape commit, this also covers the "after reshape commits" case.

**logger.cpp -- log_noexcept()/try_log() false positives**
Fixed. `Logger::log()` now returns a delivery bool: it propagates `AsyncLogger::enqueue()`'s accept/drop result on the async path, and returns the file stream's `good()` after the write (and any flush) on the sync path, returning false when the sink is closed/unhealthy. `log_noexcept()` returns that value (false on exception), and `try_log()` forwards it, so callers get an accurate success/failure indicator.

**memory.cpp -- logging inside the s_cache_state_mutex critical section**
Not changing this, because the lock-under-log deadlock it guards against is not reachable here. That deadlock needs the Logger to acquire `s_cache_state_mutex` while another thread holds it and waits on the Logger. The Logger never calls into the memory cache (it does file I/O under its own mutex), so the inversion cannot occur. These are also cold init/clear/shutdown paths, not the hot path (the hot path uses per-shard locks, not `s_cache_state_mutex`). A multi-function collect-and-emit refactor would add churn for a deadlock that cannot happen. Happy to do it purely for convention consistency if you'd prefer -- just confirm.

**memory.cpp -- pin_current_module() return value not checked before detach**
Not changing this. `pin_current_module()` pins the current module via `GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_PIN, &pin_current_module, ...)`, i.e. it resolves the module from the address of its own running code, which is loaded by definition, so it cannot realistically fail; it also already emits a diagnostic via OutputDebugStringA on the (unreachable) failure. The suggested fallback (leave the thread joinable, do not flip `s_cache_initialized`) runs under the loader lock, where joining deadlocks, so a still-joinable `std::thread` would call `std::terminate` in its destructor at teardown -- strictly worse than the current pin-then-detach, which is the correct least-bad action on the FreeLibrary path.

**input.cpp -- poll_loop and release_active_holds() use throwing Logger::error in catch handlers**
Fixed. Both the poll-thread callback handler and the noexcept `release_active_holds()` handler now use `Logger::try_log(LogLevel::Error, ...)` instead of `Logger::error(...)`, so a formatting or sink failure cannot throw out of the polling thread or out of the noexcept function.
